Classes_Homework_5/fromfile.py

```diff
@@ -0,0 +1,254 @@
+        return a_txt, a_date, formatted
+
+    def recipe_values(self, section):
+        """Extract values for Recipe section"""
+        r_txt = self.check_value(label3, section, text+':')
+        r_cal = self.check_value(label3, section, calories+':')
+        fit_tip = e.get_fitness_message(r_cal)
+        return r_txt, r_cal, fit_tip
+
+    def parse_file(self, sections):
+        """Prepare article sections for file. Return section name + attributes for each section to be late used for db"""
+        final_text = ''
+        values = []
+        for section in sections:
+            db.create_table(section.split()[0])
+            com = Combine(section.split()[0], decor, decor_length)
+            if section.split()[0] == label1:
+                n_city, n_txt, n_date = self.news_values(section)
+                final_text += com.get_news(n_city, n_txt, n_date)+'\n\n'
+                values.append([section.split()[0], n_city, n_txt, n_date])
+            elif section.split()[0] == label2:
+                a_txt, a_date, formatted = self.ad_values(section)
+                final_text += com.get_ad(a_txt, a_date, formatted)+'\n\n'
+                # check if date is not in the past
+                di.raise_if_past(a_date)
+                values.append([section.split()[0], a_txt, formatted])
+            elif section.split()[0] == label3:
+                r_txt, r_cal, fit_tip = self.recipe_values(section)
+                final_text += com.get_recipe(r_txt, r_cal, fit_tip)+'\n\n'
+                values.append([section.split()[0], r_txt, r_cal, fit_tip])
+        return final_text.rstrip(), values
+
+    def write_to_db(self, section):
+        """
+        Creates tables, inserts values, checks for duplicates, writes logs
+
+        section = list of values for each section in the following order: section label (News etc), section attributes
+
+        """
+        db.create_table(section[0])  # first element in section list is section name, like News/Ad/Recipe
+        if section[0] == label1:
+            try:
+                db.go(cnf.get_values("SQL", "insert_news"), (section[1], section[2], section[3]))
+            except sql.IntegrityError:
+                raise Duplicate(section[0], section[1], section[2])
+            except KeyError as err:
+                print(err)
+        elif section[0] == label2:  # Private ad
+            try:
+                db.go(cnf.get_values("SQL", "insert_ad"), (section[1], section[2]))
+            except sql.IntegrityError:
+                raise Duplicate(section[0], section[1], section[2])
+            except KeyError as err:
+                print(err)
+        elif section[0] == label3:  # Recipe
+            try:
+                db.go(cnf.get_values("SQL", "insert_recipe"), (section[1], section[2], section[3]))
+            except sql.IntegrityError:
+                raise Duplicate(section[0], section[1], section[2])
+            except KeyError as err:
+                print(err)
```

Comment:
This code looks very similar to code in the modules/run.py file. I think it's possible to reuse it, not declare one more time.

By the way, if you want to pass an unknown count of arguments to your function, you can use **args/kwargs** arguments:

https://medium.com/nuances-of-programming/%D1%87%D0%B5%D0%BC-%D1%8F%D0%B2%D0%BB%D1%8F%D1%8E%D1%82%D1%81%D1%8F-args-%D0%B8-kwargs-%D0%B2-python-c42adfe371bc